src/content/learn/understanding-your-ui-as-a-tree.md

```diff
@@ -0,0 +1,268 @@
+
+From components, React creates a UI tree which React DOM uses to render the DOM
+</Diagram>
+
+React also uses tree structures to manage and model the relationship between components in a React app. These trees are useful tools to understand how data flows through a React app and how to optimize rendering and app size.
+
+## The Render Tree {/*the-render-tree*/}
+
+A major feature of components is the ability to compose components of other components. As we [nest components](/learn/your-first-component#nesting-and-organizing-components), we have the concept of parent and child components, where each parent component may itself be a child of another component. 
+
+When we render a React app, we can model this relationship in a tree, known as the render tree.
+
+Here is a React app that renders a random greeting. 
+
+<Sandpack>
+
+```js App.js
+import Title from './Title';
+import Greeting from './Greeting';
+import Copyright from './Copyright';
+
+export default function App() {
+  return (
+    <>
+      <Title text="Generate a random greeting" />
+      <Greeting>
+        <Copyright year={2004} />
+      </Greeting>
+    </>
+  );
+}
+
+```
+
+```js Title.js
+export default function Title({fancy, text}) {
+    return <h1 className={fancy ? 'fancy' : ''}>{text}</h1>;
+}
+```
+
+```js Greeting.js
+import * as React from 'react';
+import greetings from './greetings';
+import Title from './Title';
+
+function randIndex(max) {
+  return Math.floor(Math.random() * max);
+}
+
+export default function Greeting({children}) {
+  const [index, setIndex] = React.useState(randIndex(greetings.length));
+  const greeting = greetings[index];
+  const nextGreeting = () => setIndex(randIndex(greetings.length));
+  return (
+    <>
+      <Title fancy text={greeting} />
+      <button onClick={nextGreeting}>🎲</button>
+      {children}
+    </>
+  );
+}
+```
+
+```js Copyright.js
+export default function Copyright({year}) {
+  return <p>©️ {year}</p>;
+}
+```
+
+```js greetings.js
+export default ['Howdy!', 'Hello2!', 'Hello3!', 'Hello3!', 'Hello4!'];
+```
+
+```css
+.fancy {
+    font-style: italic;
+    color: blue;
+}
+```
+
+</Sandpack>
+
+<Diagram name="render_tree" height={300} width={509} alt="Tree graph with 5 nodes. The root of the tree is App, with two arrows extending from it to nodes Greeting and Title. The arrows are labelled renders. Greeting node also has two arrows extended pointing down to nodes Title and Copyright.">
+
+React creates a UI tree made of components rendered, known as a render tree.
+
+</Diagram>
+
+From the example app, we can construct the above render tree. Each node in the tree represents a component and the root node is the [root component](/learn/importing-and-exporting-components#the-root-component-file). In this case, the root component is `App` and it is the first component React renders. Each arrow in the tree points from a parent component to a child component.
```

Comment:
> node

I think this is the first time we've introduced the concept of a `node`. It might be worth a breif explination and/or a link to where someone can learn more about nodes and trees. Some people reading this might be learning how to program for the first time.

Reply:
Added a bit more clarity to this